### ISSUE TITLE
Display dims while playing (screensaver suspend is on)

### ISSUE THREAD

**Issue by Bero1707** [libretro/RetroArch#5909]
Display dims although screensaver suspend option is on.
This didn't happen with version 1.6.7. Only since I upgraded to 1.6.9.
I'm using Windows 7 64 bit Ultimate.

**Comment by bparker06** [comment]
The screensaver suspend option only prevents an actual screensaver program from starting up, which is not related to any other power management technique like display brightness adjustment for example.

**Comment by Bero1707** [comment]
I never had that problem with previous versions.

**Comment by bparker06** [comment]
You would have to turn off the auto brightness adjustment within Windows, RetroArch does not control your display's brightness. However if you wanted to request a feature to be able to do so, it is possible to add it for Windows Vista and later.

**Comment by Bero1707** [comment]
This was one of many little things that brought me to RetroArch (that I don't have to change power options every time I wanted to play) cause not every emulator had screensaver/display dim & turn off suspend option. I hated that!
If you could make controller button presses be accepted by Windows, like keyboard or mouse for example, (within RetroArch) so that display never dims or turns off when playing that would be better option cause screen should dim/turn off if there is no input but forcing display to always be on even when there's no input would be good enough for me like it was until now.
Thanks!

**Comment by bparker06** [comment]
I don't think we could (or would want to) make button presses "accepted by Windows", but if you're wanting a feature to simply keep the display *on* (I thought we were talking about dimming?), that is supposed to already be supported by the 'suppress screensaver' feature. If you're saying your display is indeed turning completely *off*, then that's a separate bug that needs to be filed.

**Comment by Bero1707** [comment]
My display is set to dim if there is no input for 10 minutes and turn off after 15 minutes of no input. Dimming always precedes turning off. I thought suspend screensaver option implies this. Why wouldn't it? Please include this in next version of RetroArch.
Thanks!

**Comment by bparker06** [comment]
I'm still not sure if you're saying that your display eventually turns completely off or not while running RA. Does it just dim, or both dim + poweroff later ?

**Comment by Bero1707** [comment]
Yes. It does turn off completely! It dims after 10 minutes of gameplay and then after 5 more minutes it turns off completely but I usually move mouse a bit or press something on keyboard when it dims to prevent it from completely turning off.

**Comment by bparker06** [comment]
Yea that's not supposed to happen. I just tried it here after setting the display timeout to 1 minute and RA kept the display on as it should... so I'm not able to reproduce your issue.

**Comment by Bero1707** [comment]
OK I tested this again. I set "Dim the display" to 1 min and "Turn off display" to 2 min. I didn't move mouse or press any keyboard key after display dimmed after 1 min (I always did that to un-dim the display before). I played with my controller as usual and display didn't turn off after all.
Can you please include suspending display dimming within "Suspend screensaver" option to prevent dimming as well as screensaver and turning off?

**Comment by bparker06** [comment]
Unfortunately I don't see any documentation on Microsoft's site that explicitly tells me what might disable screen dimming. The function we use to prevent the screen turning off is here:

https://msdn.microsoft.com/en-us/library/windows/desktop/dd405534(v=vs.85).aspx

Specifically the `PowerRequestDisplayRequired` type as you can see in the code here:

https://github.com/libretro/RetroArch/blob/709c7de40fdcab0d968d454e6aa2c2614bd5799e/gfx/common/win32_common.c#L919

**Comment by Bero1707** [comment]
OK then I'll disable display dimming for now, as I found out that display doesn't turn off after all, until you find the solution for dimming, if it's possible.
Thank you very much!

**Comment by ofry** [comment]
https://msdn.microsoft.com/en-us/library/windows/desktop/aa373208(v=vs.85).aspx

**Comment by ofry** [comment]
@bparker06 Is this article help?

**Comment by bparker06** [comment]
No, and we don't use that function for Windows 7 and up... we use [PowerSetRequest](https://msdn.microsoft.com/en-us/library/windows/desktop/dd405534(v=vs.85).aspx) there.

**Comment by ofry** [comment]
Maybe to solve this bug you should use this function?

**Comment by fr500** [comment]
Hmmm, windows does prevent dimming / poweroff when playing games automatically IF YOU'RE USING XINPUT pads.

With DINPUT it's not preveneted.

**Comment by Bero1707** [comment]
That's right! I'm using DINPUT controller adapter with Dualshock2.

**Comment by KingDuckZ** [comment]
I have the same problem on KDE Gentoo. I'm using udev as the input driver. Retroarch 1.8.5.

Please not that mpv and vlc, for example, do prevent the screen from getting dimmed even if there is no input while I watch movies. I suggest Retroarch should do the same if you're in-game, and dim on no input if you're in the menu, or maybe there should be an option to control this behaviour.

**Comment by guoyunhe** [comment]
This is clearly a duplicate of #7472 

`ScreenSaver` DBus API prevent both screen locking and screen dim in KDE and GNOME. If your screen dim, it means RetroArch isn't build with `--enable-dbus` flag. And you should ask the maintainer of distribution you use to update the package build script.
